[test] Isolate `build-script` calls in `skip-local-build.test-sh`

edymtt opened the pull request:
...and reenable `llvm-targets-options.test` (previously disabled in #37573).

This will align `skip-local-build.test-sh` with the behaviour of the
other BuildSystem tests, by

* ensuring we use the `cmake` exposed in `lit.cfg`, so that under Linux we
don't attempt to rebuild it
* using a separate build folder for `build-script` invocations, so that
side effects will not affect the main invocation and other lit tests.

I expect these changes to prevent `llvm-targets-options.test` to fail in Linux
presets with an error related to cmake, e.g.

```
build-script: error: argument --cmake: /home/buildnode/jenkins/workspace/
oss-swift-package-linux-ubuntu-18_04/build/cmake-linux-x86_64/bin/cmake is not an executable
```

Addresses rdar://78320684

edymtt commented:
@swift-ci please smoke test

benlangmuir commented:
@edymtt is there a way to test the original issue in PR?  I assume regular testing didn't cover it or you would have caught it before merging.  Change LGTM

edymtt commented:
@benlangmuir not at the moment, since I don't know why the permissions on that file changed in the first place -- in my investigation I was not able to reproduce this bug, even when running `llvm-targets-options.test` and `skip-local-build.test-sh` in parallel (the final cmake executable was never updated, only a bunch of intermediate files were)

So before introducing logic for a failure I don't fully understand, I preferred to focus first on reducing any chance for existing tests to rebuild CMake to narrow down the scope (and in part to leverage the scale on the CI should this reoccur)

porglezomp commented on validation-test/BuildSystem/skip-local-build.test-sh:
```diff
@@ -1,8 +1,12 @@
-# RUN: %swift_src_root/utils/build-script --dump-config --skip-local-build 2>&1 | %FileCheck %s -check-prefix=CONFIG
+
+# RUN: %empty-directory(%t)
+# RUN: mkdir -p %t
```
`%empty-directory(%t)` expands to `rm -rf %t && mkdir -p %t` so this is redundant.

edymtt replied:
Thanks for noticing this!

Tackled in 7379f455bf06d3e29ed895256f3b622fde0d50bc -- took the chance to rebase the PR so that there is a separate commit to reenable `lvm-targets-options.test`

edymtt committed:
[test] Isolate build-script calls in skip-local-build.test-sh

This will align skip-local-build.test-sh with the behaviour of the
other BuildSystem tests, by

* ensuring we use the cmake exposed in lit.cfg, so that under Linux we
don't attempt to rebuild it
* using a separate build folder for build-script invocations, so that
side effects will not affect the main invocation and other lit tests.

I expect these changes to prevent llvm-targets-options.test to fail in
Linux presets with an error related to cmake, e.g.

```
build-script: error: argument --cmake:
/home/buildnode/jenkins/workspace/
oss-swift-package-linux-ubuntu-18_04/build/cmake-linux-x86_64/bin/cmake
is not an executable
```

Addresses rdar://78320684

edymtt committed:
Reenable `llvm-targets-options.test`

This was previously disabled in #37573.

edymtt commented:
@swift-ci please smoke test

edymtt commented:
@swift-ci please smoke test macOS
